flink-runtime/src/main/java/org/apache/flink/runtime/source/coordinator/SourceCoordinator.java

```diff
@@ -270,8 +269,9 @@ public void start() throws Exception {
     @Override
     public void close() throws Exception {
         LOG.info("Closing SourceCoordinator for source {}.", operatorName);
+        closeQuietly(context);
         if (started) {
-            closeAll(asList(context, enumerator), Throwable.class);
+            closeQuietly(enumerator);
```

Comment:
Could we force constraints on the order of closing `context` and `enumerator` as follows:
Because `context` was used by  `enumerator`, This is a bit like the close order in the I/O mechanism.

```
        if (started) {
            closeQuietly(enumerator);
        }
        closeQuietly(context);
```

Although not doing so will not result in semantic errors.

Reply:
Thanks for the review, good catch! updated.